### PR TITLE
test(unit): unit tests with Jest

### PR DESCRIPTION
**Pull request by barsheshet**
* Introducing Jest
* index unit tests
* namespace unit tests
* upgrade "request" to fix known vulnerabilities

### CODE REVIEW

**Comment by barsheshet** [comment]
@guyguyon BTW,

This is me ,Guy, from "Shevet Rehasim" if you remember... 
You have no public email so i couldn't reach you.

File: spec/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      expect(asyncHooks.createHook).toBeCalledTimes(1);
+      expect(enableSpy).toBeCalledTimes(1);
+    });
+  });
+
+  describe('Get namespace tests', () => {
+    it('should get the created namespace', () => {
+      asyncHooks.createHook.mockReturnValueOnce({ enable: () => true });
+      const yam = createNamespace('yam');
+      const yabasha = getNamespace('yam');
```
Comment (guyguyon) [inline spec/index.spec.js]: hahaha

**Comment by guyguyon** [comment]
@barsheshet What's up! long time....
Your PR is great. Thanks.
If you want, you can add some e2e tests. 
Something with a real async operation.
Let me know if you want, otherwise, i'll merge it

**Comment by barsheshet** [comment]
Sure. I'll add integration/e2e tests in a different PR when i find the time.
Meanwhile you can go ahead and merge it.

**Pull request merged**